### PR TITLE
feat(ui): build device summary page

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done

- Built device summary page. The design is a bit different to legacy to make it look more like the machine summary.
- Made the machine summary network card reusable across node types and improved responsiveness

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to /MAAS/r/devices and click on a device
- Check that you are taken to the summary tab and all the data is correct
- Check that the page looks alright on all screen sizes

## Fixes

Fixes canonical-web-and-design/app-tribe#536

### Large screen
Screenshot 2021-12-03 at 16-49-47 test summary bolla MAAS

### Small screen
Screenshot 2021-12-03 at 16-56-07 test summary bolla MAAS

### CODE REVIEW

**Comment by Caleb-Ellis** [comment]
@amylily1011 @rayito let me know if this looks alright to you, I just kinda winged it to look a bit closer to the machine summary.

File: ui/src/app/base/components/NodeSummaryNetworkCard/NodeSummaryNetworkCard.test.tsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
+import { mount } from "enzyme";
```
Comment (Caleb-Ellis) [inline ui/src/app/base/components/NodeSummaryNetworkCard/NodeSummaryNetworkCard.test.tsx:1]: This looks like a large diff but it's mainly moving over the file from the machine summary to base components

**Comment by amylily1011** [comment]
Love it, works for me, Amy +1.

File: ui/src/app/machines/views/MachineDetails/MachineSummary/MachineSummary.tsx [hunk 2]
```diff
@@ -39,6 +44,7 @@ const MachineSummary = ({ setHeaderContent }: Props): JSX.Element => {
     return <Spinner text="Loading" />;
   }
 
+  const networkURL = machineURLs.machine.network({ id });
```
Comment (huwshimi) [inline ui/src/app/machines/views/MachineDetails/MachineSummary/MachineSummary.tsx:47]: Thanks, this is better than the hardcoded URL we had before.

**Review by huwshimi — APPROVED** [review]

**Pull request merged**